praw/models/util.py

```diff
@@ -13,19 +14,23 @@ class BoundedSet:
     def __init__(self, max_items: int):
         """Construct an instance of the BoundedSet."""
         self.max_items = max_items
-        self._fifo = []
-        self._set = set()
+        self._set = OrderedDict()
 
     def __contains__(self, item: Any) -> bool:
         """Test if the BoundedSet contains item."""
+        self._access(item)
         return item in self._set
 
+    def _access(self, item: Any):
+        if item in self._set:
+            self._set.move_to_end(item)
+
     def add(self, item: Any):
         """Add an item to the set discarding the oldest item if necessary."""
-        if len(self._set) == self.max_items:
-            self._set.remove(self._fifo.pop(0))
-        self._fifo.append(item)
-        self._set.add(item)
+        self._access(item)
+        self._set[item] = None
+        while len(self._set) > self.max_items:
```

Comment:
Why is a while loop utilized here? Won't this always execute once?

Reply:
You're right, and it should be an `if` statement. 
```suggestion
        if len(self._set) > self.max_items:
```

Reply:
I don't feel strongly about it, but FWIW I made it a while loop just for the sake of defensive programming: it *shouldn't* be called more than once, but 1) if it is called more than once then it's still correct and 2) in case the bound invariant is somehow violated, the `while` loop will put it back in order.